sklearn/compose/_target.py

```diff
@@ -184,7 +190,7 @@ def fit(self, X, y, sample_weight=None):
             self.regressor_ = clone(self.regressor)
 
         # transform y and convert back to 1d array if needed
-        y_trans = self.transformer_.fit_transform(y_2d)
+        y_trans = self.transformer_.transform(y_2d)
```

Comment:
Moving the following block to up a few lines so it follows `self._fit_transformer(y_2d)` would make it easier to follow how `y` transforms.

```python
y_trans = self.transformer_.transform(y_2d)
...
if y_trans.ndim == 2 and y_trans.shape[1] == 1:
    y_trans = y_trans.squeeze(axis=1)
```